producer/src/contract/contract_test.go

```diff
@@ -136,184 +137,230 @@ func TestYieldSerialization(t *testing.T) {
-	err := InsertYield(testYieldFloor, "testDB.dat", 35, contractHashFloor, 1)
+func TestMakeClaimMultipleYield(t *testing.T) {
+	db := "testDB.db"
+	setUpDB(db)
+	defer tearDown(db)
+
+	amt := []uint64{uint64(100), uint64(200)}
+	pubKey := []ecdsa.PublicKey{generatePubKey(), generatePubKey()}
+	yield := []Yield{MakeYield(&pubKey[0], amt[0]), MakeYield(&pubKey[1], amt[1])}
+	contractHash := [][]byte{block.HashSHA256([]byte("1")), block.HashSHA256([]byte("2"))}
+	blockHeight := []uint64{uint64(35), uint64(40)}
+	yieldIndex := []uint16{uint16(1), uint16(4)}
+
+	err := InsertYield(yield[0], db, blockHeight[0], contractHash[0], yieldIndex[0])
 	if err != nil {
-		t.Errorf("Failed to insert yield")
+		t.Errorf("Failed to insert yield: %v", err)
 	}
-	err = InsertYield(testYieldFloor, "testDB.dat", 35, contractHashCeiling, 1)
+	err = InsertYield(yield[1], db, blockHeight[1], contractHash[1], yieldIndex[1])
 	if err != nil {
-		t.Errorf("Failed to insert yield")
+		t.Errorf("Failed to insert yield: %v", err)
 	}
-	testClaim, err := MakeClaim("testDB.dat", testPubKey, 10000008)
+
+	testClaim, err := MakeClaim(db, pubKey[1], uint64(50))
 	if err == nil {
-		t.Errorf("Should have an error that resembles change")
+		t.Errorf("No change error struct returned")
+	} else {
+		if change, ok := err.(ChangeError); ok {
+			if change.Change != uint64(150) {
+				t.Errorf("Wrong change value, expected: 5, actual: %d", change)
+			}
+		} else {
+			t.Errorf("Different error given: %v", err)
+		}
 	}
-	if !bytes.Equal(testClaim.PreviousContractHash, contractHashCeiling) {
+
+	if !bytes.Equal(testClaim.PreviousContractHash, contractHash[1]) {
 		t.Errorf("Contract hashes do not match")
 	}
-	if testClaim.BlockIndex != 35 {
+	if testClaim.BlockIndex != blockHeight[1] {
 		t.Errorf("Block indeces do not match")
 	}
-	if testClaim.YieldIndex != 1 {
+	if testClaim.YieldIndex != yieldIndex[1] {
 		t.Errorf("Yield indeces do not match")
 	}
-	if !cmp.Equal(testClaim.PublicKey, testPubkey) {
+	if !reflect.DeepEqual(testClaim.PublicKey, pubKey[1]) {
 		t.Errorf("Public Keys do not match")
 	}
+
 }
 
-func TestMakeClaimPriorityCase2(t *testing.T) {
-	setUpDB("testDB.db")
-	defer tearDown("testDB.db")
-	testPubKey := generatePubKey()
-	testYieldFloor := MakeYield(testPubKey, 10000000)
-	testYieldCeiling := MakeYield(testPubKey, 10000010)
-	contractHashFloor := block.HashSHA256([]byte("This should be claimed"))
-	contractHashCeiling := block.HashSHA256([]byte("This should not be claimed"))
-	err := InsertYield(testYieldFloor, "testDB.dat", 35, contractHashFloor, 1)
-	if err != nil {
-		t.Errorf("Failed to insert yield")
-	}
-	err = InsertYield(testYieldFloor, "testDB.dat", 35, contractHashCeiling, 1)
+func TestMakeClaimEmpty(t *testing.T) {
+	db := "testDB.db"
+	setUpDB(db)
+	defer tearDown(db)
+
+	amt := []uint64{uint64(100), uint64(200)}
+	pubKey := []ecdsa.PublicKey{generatePubKey(), generatePubKey(), generatePubKey()}
```

Comment:
multiple public keys?

Reply:
Yeah there will be yields corresponding to different public keys in the database.

Reply:
There can be in the case of the producer, but that wasn't the original intent behind this test case.